packages/kbn-esql-utils/src/utils/esql_fields_utils.ts

```diff
@@ -40,22 +41,30 @@ export const isESQLColumnSortable = (column: DatatableColumn): boolean => {
   return true;
 };
 
+// Helper function to check if a field is groupable based on its type and esType
+const isGroupable = (type: string | undefined, esType: string | undefined): boolean => {
+  if (type === UNKNOWN_FIELD) {
+    return false;
+  }
+  if (esType && esType.indexOf(TSDB_COUNTER_FIELDS_PREFIX) !== -1) {
+    return false;
+  }
+  return true;
+};
+
 /**
  * Check if a column is groupable (| STATS ... BY <column>).
  *
  * @param column The DatatableColumn of the field.
  * @returns True if the column is groupable, false otherwise.
  */
-
 export const isESQLColumnGroupable = (column: DatatableColumn): boolean => {
-  // we don't allow grouping on the unknown field types
-  if (column.meta?.type === UNKNOWN_FIELD) {
-    return false;
-  }
-  // we don't allow grouping on tsdb counter fields
-  if (column.meta?.esType && column.meta?.esType?.indexOf(TSDB_COUNTER_FIELDS_PREFIX) !== -1) {
-    return false;
-  }
+  return isGroupable(column.meta?.type, column.meta?.esType);
+};
 
-  return true;
+export const isESQLFieldGroupable = (field: FieldSpec): boolean => {
+  const isCounterTimeSeries = field.timeSeriesMetric === 'counter';
```

Comment:
I am again (sorry) very sceptical for this function.
As long as this `isCounterTimeSeries = field.timeSeriesMetric === 'counter';` is true you should return false, no need to make the concatenation. 

So the only thing that needs to be tested is this

```
if (type === UNKNOWN_FIELD) {
    return false;
  }
```
cc @jughosta wdyt?

(also waiting to see if the FieldSpec import will influence negatively the package size)

Reply:
ok the stats are good! Can you not do the concatenation above and return? is not needed